### PR TITLE
disttask: cleanup framework meta when nodes shutdown

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #46258 

Problem Summary:

### What is changed and how it works?
1. Compare nodes stored in infosync and dist_framework_meta when doing cleanUpRoutine.
Then delete legacy nodes.
2. Mock taskManager.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by wjhuang2016 — APPROVED** [review]

File: pkg/disttask/framework/dispatcher/dispatcher.go [hunk 1]
```diff
@@ -137,17 +137,12 @@ func (*BaseDispatcher) Close() {
...
-	newTask, err := d.taskMgr.GetGlobalTaskByID(d.Task.ID)
+func (d *BaseDispatcher) refreshTask() (err error) {
+	d.Task, err = d.taskMgr.GetGlobalTaskByID(d.Task.ID)
 	if err != nil {
 		logutil.Logger(d.logCtx).Error("refresh task failed", zap.Error(err))
-		return err
-	}
-	// newTask might be nil when GC routine move the task into history table.
-	if newTask != nil {
-		d.Task = newTask
```
Comment (D3Hunter) [inline pkg/disttask/framework/dispatcher/dispatcher.go:148]: why removed?

File: pkg/disttask/framework/dispatcher/dispatcher_manager.go [hunk 2]
```diff
@@ -274,13 +274,10 @@ func (dm *Manager) startDispatcher(task *proto.Task) {
 			dm.failTask(task, err)
 			return
 		}
-		defer func() {
-			dispatcher.Close()
-			dm.delRunningTask(task.ID)
-		}()
```
Comment (D3Hunter) [inline pkg/disttask/framework/dispatcher/dispatcher_manager.go:280]: here too

File: pkg/disttask/framework/dispatcher/dispatcher_manager.go [hunk 3]
```diff
@@ -334,6 +335,36 @@ func (dm *Manager) doCleanUpRoutine() {
...
+		return 0
+	}
+
+	cleanNodes := make([]string, 0)
+	for _, nodeID := range oldNodes {
+		if ok := disttaskutil.MatchServerInfo(serverInfos, nodeID); !ok {
+			cleanNodes = append(cleanNodes, nodeID)
+		}
+	}
+	logutil.BgLogger().Info("start to clean up dist_framework_meta")
```
Comment (D3Hunter) [inline pkg/disttask/framework/dispatcher/dispatcher_manager.go:359]: return when no nodes to clean, to avoid log too much

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

**Pull request merged**